src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace phpDocumentor;
+
+use phpDocumentor\ConfigurationFactory\PhpDocumentor2;
+use phpDocumentor\ConfigurationFactory\PhpDocumentor3;
+use phpDocumentor\ConfigurationFactory\Strategy;
+
+final class ConfigurationFactory
+{
+    /**
+     * The Uri that contains the path to the configuration file
+     *
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * The configuration xml
+     *
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * The path to the xsd that is used for validation of the configuration file.
+     *
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * All strategies that are used by the ConfigurationFactory
+     *
+     * @var Strategy[]
+     */
+    private $strategies;
+
+    /**
+     * Initializes the ConfigurationFactory.
+     *
+     * @param Uri    $uri
+     * @param string $schemaPath
+     */
+    public function __construct(Uri $uri, $schemaPath = '')
+    {
+        if ($schemaPath === '') {
```

Comment:
Schema path is a phpdoc3 config thing only. So that should be injected into to the strategy. Since it may defer per version.